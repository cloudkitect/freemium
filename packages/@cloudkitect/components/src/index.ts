export * from './ck-certificatemanager/ck-dns-validated-certificate';
export * from './ck-cloudfront/ck-distribution';
export * from './ck-core/ck-vendor-tags';
export * from './ck-core/ck-tag';
export * from './ck-core/ck-utils';
export * from './ck-core/ck-region';
export * from './ck-dynamodb/ck-table-v2';
export * from './ck-ec2/ck-vpc';
export * from './ck-ecr/ck-repository';
export * from './ck-ecs/ck-cluster';
export * from './ck-ecs/ck-fargate-clsuter';
export * from './ck-ecs/ck-fargate-service';
export * from './ck-ecs/ck-fargate-task-definition';
export * from './ck-elasticloadbalancingv2/ck-public-application-loadbalancer';
export * from './ck-s3/ck-bucket';
export * from './ck-stack/ck-stack';
