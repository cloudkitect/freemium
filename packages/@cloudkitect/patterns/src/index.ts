export * from './ck-website/ck-serverless-website';
export * from './ck-container/ck-serverless-container-app';