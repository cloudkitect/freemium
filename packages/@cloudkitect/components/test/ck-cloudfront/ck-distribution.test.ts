/*
 * Copyright (c) CloudKitect Inc. All Rights Reserved.
 */
import { App } from 'aws-cdk-lib';
import { Match, Template } from 'aws-cdk-lib/assertions';
import {
  SecurityPolicyProtocol,
  ViewerProtocolPolicy,
} from 'aws-cdk-lib/aws-cloudfront';
import { HttpOrigin } from 'aws-cdk-lib/aws-cloudfront-origins';
import {
  CkAccountType, CkStack,
} from '../../lib';
import { CkDistribution } from '../../src';

let app: App;
let stack: CkStack;
let testOrigin: HttpOrigin;

const stackProps = {
  ckAccountType: CkAccountType.PRODUCTION,
  ckApplication: 'Test',
  ckCompany: 'CloudKitect',
};

describe('Well Architected CloudFront Distribution', () => {
  beforeEach(() => {
    stack = new CkStack(app, 'TestStack', { ...stackProps });
    testOrigin = new HttpOrigin('fake.example.com');
  });

  test('Default cloudfront settings', () => {
    new CkDistribution(stack, 'TestWaDistribution', {
      defaultBehavior: { origin: testOrigin },
    });

    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });

  test.each`
      actual                                  | expected
      ${undefined}                            | ${'TLSv1.2'}
      ${null}                                 | ${'TLSv1.2'}
      ${SecurityPolicyProtocol.TLS_V1_2_2021} | ${'TLSv1.2'}
      ${SecurityPolicyProtocol.TLS_V1_2_2019} | ${'TLSv1.2'}
   `(
    'CC-313: MinimumProtocolVersion should be $expected when $actual',
    ({ actual, expected }) => {
      new CkDistribution(stack, 'TestWaDistribution', {
        defaultBehavior: { origin: testOrigin },
        minimumProtocolVersion: actual,
      });
      const template = Template.fromStack(stack);

      template.hasResourceProperties('AWS::CloudFront::Distribution', {
        DistributionConfig: Match.objectLike({
          Origins: [
            {
              CustomOriginConfig: {
                OriginProtocolPolicy: 'https-only',
                OriginSSLProtocols: [expected],
              },
              DomainName: 'fake.example.com',
              Id: Match.stringLikeRegexp(
                'TestStackTestWaDistributionOrigin*',
              ),
            },
          ],
        }),
      });
    },
  );

  test.each`
      actual                                    | expected
      ${undefined}                              | ${'redirect-to-https'}
      ${null}                                   | ${'redirect-to-https'}
      ${ViewerProtocolPolicy.REDIRECT_TO_HTTPS} | ${'redirect-to-https'}
      ${ViewerProtocolPolicy.ALLOW_ALL}         | ${'allow-all'}
   `(
    'CC-312: REDIRECT_TO_HTTPS viewer protocol should be $expected when $actual',
    ({ actual, expected }) => {
      new CkDistribution(stack, 'TestWaDistribution', {
        defaultBehavior: {
          origin: testOrigin,
          viewerProtocolPolicy: actual,
        },
      });
      const template = Template.fromStack(stack);
      template.hasResourceProperties('AWS::CloudFront::Distribution', {
        DistributionConfig: {
          DefaultCacheBehavior: Match.objectLike({
            ViewerProtocolPolicy: expected,
          }),
        },
      });
    },
  );

});
