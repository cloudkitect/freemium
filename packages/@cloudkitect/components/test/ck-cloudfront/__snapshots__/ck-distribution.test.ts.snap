// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Well Architected CloudFront Distribution Default cloudfront settings 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "TestWaDistributionB5310A2E": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "DistributionConfig": {
          "DefaultCacheBehavior": {
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "TargetOriginId": "TestStackTestWaDistributionOrigin17582B37B",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": [
            {
              "CustomOriginConfig": {
                "OriginProtocolPolicy": "https-only",
                "OriginSSLProtocols": [
                  "TLSv1.2",
                ],
              },
              "DomainName": "fake.example.com",
              "Id": "TestStackTestWaDistributionOrigin17582B37B",
            },
          ],
          "PriceClass": "PriceClass_All",
        },
        "Tags": [
          {
            "Key": "CkComponents",
            "Value": "2.128.0",
          },
          {
            "Key": "CkVendor",
            "Value": "CloudKitect",
          },
        ],
      },
      "Type": "AWS::CloudFront::Distribution",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
